pkg/actions/param_list.go

```diff
@@ -62,10 +72,17 @@ func NewParamList(m map[string]interface{}) (*ParamList, error) {
 		componentName: ol.LoadOptionalString(OptionComponentName),
 		envName:       ol.LoadOptionalString(OptionEnvName),
 
-		out:             os.Stdout,
-		findModulesFn:   component.ModulesFromEnv,
-		findModuleFn:    component.GetModule,
-		findComponentFn: component.LocateComponent,
+		out:          os.Stdout,
+		findModuleFn: component.GetModule,
+	}
+
+	if pl.app != nil {
+		p := pipeline.New(pl.app, pl.envName)
+		pl.modulesFn = p.Modules
```

Comment:
We treat app as optional here by not returning an error, but depend on `pl.modulesFn`, `pl.envParametersFn`, and `pl.lister` being non-nil below.

Reply:
Yes, but the `ol.err` check below will bail out if app is not set. So the assumption is if you got this far, everything was set properly.

Reply:
In that case maybe reordering to check `ol.err` before accessing `pl.app` here would be clearer?